xdsl/dialects/builtin.py

```diff
@@ -2274,6 +2274,24 @@ def iter_values(self) -> Iterator[int] | Iterator[float]:
         """
         return self.get_element_type().iter_unpack(self.data.data)
 
+    def get_int_values(self) -> Sequence[int]:
+        """
+        Return all the values of the elements in this DenseIntOrFPElementsAttr,
+        checking that the elements are integers.
+        """
+        el_type = self.get_element_type()
+        assert isinstance(el_type, IntegerType | IndexType)
+        return el_type.unpack(self.data.data, len(self))
+
+    def get_float_values(self) -> Sequence[float]:
+        """
+        Return all the values of the elements in this DenseIntOrFPElementsAttr,
+        checking that the elements are floats.
+        """
+        el_type = self.get_element_type()
+        assert isinstance(el_type, AnyFloat)
+        return el_type.unpack(self.data.data, len(self))
+
```

Comment:
A counter-proposal:
```suggestion
    @overload
    def get_values(self: DenseIntOrFPElementsAttr[IntegerType | IndexType]) -> Sequence[int]:
        
    @overload
    def get_values(self: DenseIntOrFPElementsAttr[AnyFloat]) -> Sequence[float]:
        ...
```

Reply:
I'm pretty sure I tried this and it gets weird with https://github.com/microsoft/pyright/issues/10232

Reply:
I agree that this is the ideal

Reply:
I have wrestled with Pyright and declare myself defeated.

Reply:
What I'm really confused by is why it works for ComplexType